### PR TITLE
fix(@angular-devkit/build-angular): allow function in filename while …

### PR DESCRIPTION
**Pull request by sacgrover**
…changing the name of compiling chunks

Fixes angular#17087

### CODE REVIEW

File: packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts [hunk 1]
```diff
@@ -149,7 +149,9 @@ export function getCommonConfig(wco: WebpackConfigOptions): Configuration {
                 // tslint:disable-next-line: no-any
                 (compilation.mainTemplate.hooks as any).assetPath.tap(
                   'build-angular',
-                  (filename: string, data: ChunkData) => {
+                // tslint:disable-next-line: no-any
+                  (filename: any, data: ChunkData) => {
```
Comment (clydin) [inline packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts]:
```suggestion
                  (filename: string | ((data: ChunkData) => string), data: ChunkData) => {
```

File: packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts [hunk 2]
```diff
@@ -149,7 +149,9 @@ export function getCommonConfig(wco: WebpackConfigOptions): Configuration {
...
+                    if (typeof filename === 'function') { filename = filename(data); }
```
Comment (clydin) [inline packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts]: Can you rework this so it does not reassign the argument?

Comment (sacgrover) [reply packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts]: Sure

File: packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts (hunk 1 above)
Comment (sacgrover) [reply packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts]: Thanks for this.. I had no idea that we can declare function type this way too!! noted for future!

**Comment by sacgrover** [comment]
@clydin is there any way to retry above fail checks without new commit?

**Comment by dgp1130** [comment]
Also merged to 9.0.x patch branch.

**Pull request merged**